debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/launchConfigurations/LaunchConfigurationsDialog.java

```diff
@@ -1374,9 +1375,13 @@ public void run(boolean fork, boolean cancelable, IRunnableWithProgress runnable
 			if (cancelable) {
 				fProgressMonitorPart.attachToCancelComponent(null);
 			}
-			fProgressMonitorPart.getParent().setVisible(true);
+
 			fActiveRunningOperations++;
 
+			UIJob showProgressMonitorPart = createJobToShowProgressMonitorPart();
+			// only show the progress bar if necessary after 1 second
+			showProgressMonitorPart.schedule(1_000);
```

Comment:
Since potentially not everybody wants a delay, maybe the delay should be configurable in the context?

Reply:
Do you mean like `DebugUIPlugin.getDefault().getPreferenceStore().getLong("somelong");` ? That would also mean that there needs to be a new preference in the *Preferences* dialog under *Run/Debug > Launching > Launch configurations*, right? 

Or did you mean something else?

Reply:
> Since potentially not everybody wants a delay, maybe the delay should be configurable in the context?

I was thinking again about making it configurable and it still doesn't feel right. 

Don't get me wrong, I'm all for customizing the user experience but in this particular case, a low value would mean that the progress bar might flicker and a high value would mean the UI might freeze. Letting the user set this value feels like giving him/her the choice between bad and worse 😅

Maybe you and I can settle for a value and let it hard-coded. Say 500 ms?

Reply:
There is definitely a tradeoff with configuration preferences that users are unlikely to find and as there are more and more, the user is even less likely to find and understand such things...

Reply:
Sorry for the late reply.
I meant not configurable for the user but for developers calling this code.

If a dev knows that an operation will definitvly take some time, it would be good to show i immediatly.

Furthermore If we could somehow estimate the remaining runtime(e.g. from a ProgressMonitor, it would be good to consider if it is worth to show it.
E.g. for a 501ms running Task showing Progress after 500ms is pointless.

Reply:
> I meant not configurable for the user but for developers calling this code.
Thank you for clarifying it :-) . Do you have any suggestion (code) on how to achieve that?

> Furthermore If we could somehow estimate the remaining runtime(e.g. from a ProgressMonitor, it would be good to consider if it is worth to show it.

Hm, the problem is that the progress monitor does not have any method to see how much work is still to be done so there is no basis to estimate if it's worth showing it.

>  E.g. for a 501ms running Task showing Progress after 500ms is pointless.
Agreed. The timeout is just a simple way of catching all jobs that run in under _X_ ms. I have no way of knowing how long a job is going to take, but In my experience if the job takes > 1s then it will probably take >2 s (i.e. the progress bar will appear for at least a second). Again, this is a very simplistic approach based on my experience. I wanted to provide a solution without burdening the user (or programmer) with the

Reply:
> Not really, I would guess that this should somehow be stored in the context? 

Sorry to insist but I don't understand. What _context_ do you mean?

Surely not the `org.eclipse.e4.core.contexts.IEclipseContext`, right?
![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/04ee124c-5547-4bc5-afb8-a8f399135f6e)

Reply:
Ah, you probably meant the `BundleContext` right? I'll look into it on Monday.

Reply:
> Ah, you probably meant the `BundleContext` right? I'll look into it on Monday.

I think it should be a `org.eclipse.jface.operation.IRunnableContext`, as you are using that in https://github.com/eclipse-pde/eclipse.pde/pull/674/ to show the pop-up that shows the progress indication.
Unfortunately the progress-indicator is not created directly so have to pass trough the threshold value.

Reply:
Btw. Monday is the last day of development for the upcomming 2023-09 release and I probably won't have the time to review/try this Monday evening (if another Committer does it is of course welcome).

I'll try to review your PR in PDE over the weekend, but cannot promise it.